resolver-dns/src/test/java/io/netty/resolver/dns/DnsNameResolverClientSubnetTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver.dns;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+import io.netty.handler.codec.dns.DefaultDnsOptEcsRecord;
+import io.netty.handler.codec.dns.DnsRecord;
+import io.netty.util.concurrent.Future;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class DnsNameResolverClientSubnetTest {
+
+    // See https://www.gsic.uva.es/~jnisigl/dig-edns-client-subnet.html
+    // Ignore as this needs to query real DNS servers.
+    @Ignore
+    @Test
+    public void testSubnetQuery() throws Exception {
+        EventLoopGroup group = new NioEventLoopGroup(1);
+        DnsNameResolver resolver = newResolver(group).build();
+        try {
+            // Same as:
+            // # /.bind-9.9.3-edns/bin/dig @ns1.google.com www.google.es +client=157.88.0.0/24
+            Future<List<InetAddress>> future = resolver.resolveAll("www.google.es",
```

Comment:
@Scottmitch that is why it is marked as `@Ignore`. There is currently no known java based dns server that support it and so will allow us to test it like this. I put it mainly here as an example how to use it. The encoder itself is tested in `DefaultDnsRecordEncoderTest`.